tests/api.c

```diff
@@ -33265,7 +33265,753 @@ static void test_wolfSSL_EVP_PKEY_assign(void)
+    EVP_PKEY_free(pkey);
+
+    printf(resultFmt, passed);
+#endif
+}
+
+static void test_wolfSSL_EVP_CIPHER_CTX_iv_length(void)
+{
+#if defined(OPENSSL_ALL)
+
+    /* This is large enough to be used for all key sizes */
+    byte key[AES_256_KEY_SIZE] = {0};
+    byte iv[AES_BLOCK_SIZE] = {0};
+    int i, enumlen;
+    EVP_CIPHER_CTX *ctx;
+    const EVP_CIPHER *init;
+
+    int enumArray[] = {
+
+    #ifdef HAVE_AES_CBC
+         NID_aes_128_cbc,
+    #endif
+    #if (!defined(HAVE_FIPS) && !defined(HAVE_SELFTEST)) || \
+        (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION > 2))
+    #ifdef HAVE_AESGCM
+         NID_aes_128_gcm,
+    #endif
+    #endif /* (HAVE_FIPS && !HAVE_SELFTEST) || HAVE_FIPS_VERSION > 2 */
+    #ifdef WOLFSSL_AES_COUNTER
+         NID_aes_128_ctr,
+    #endif
+    #ifndef NO_DES3
+         NID_des_cbc,
+         NID_des_ede3_cbc,
+    #endif
+    #ifdef HAVE_IDEA
+         NID_idea_cbc,
+    #endif
+    };
+    int iv_lengths[] = {
+
+    #ifdef HAVE_AES_CBC
+         AES_BLOCK_SIZE,
+    #endif
+    #if (!defined(HAVE_FIPS) && !defined(HAVE_SELFTEST)) || \
+        (defined(HAVE_FIPS_VERSION) && (HAVE_FIPS_VERSION > 2))
+    #ifdef HAVE_AESGCM
+         GCM_NONCE_MID_SZ,
+    #endif
+    #endif /* (HAVE_FIPS && !HAVE_SELFTEST) || HAVE_FIPS_VERSION > 2 */
+    #ifdef WOLFSSL_AES_COUNTER
+         AES_BLOCK_SIZE,
+    #endif
+    #ifndef NO_DES3
+         DES_BLOCK_SIZE,
+         DES_BLOCK_SIZE,
+    #endif
+    #ifdef HAVE_IDEA
+         IDEA_BLOCK_SIZE,
+    #endif
+    };
+
+
+    printf(testingFmt, "wolfSSL_EVP_CIPHER_CTX_iv_length");
+    enumlen = (sizeof(enumArray)/sizeof(int));
+    for(i = 0; i < enumlen; i++)
+    {
+        ctx = EVP_CIPHER_CTX_new();
+        init = wolfSSL_EVP_get_cipherbynid(enumArray[i]);
+
+        wolfSSL_EVP_CIPHER_CTX_init(ctx);
+        AssertIntEQ(EVP_CipherInit(ctx, init, key, iv, 1), WOLFSSL_SUCCESS);
+
+        AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_iv_length(ctx), iv_lengths[i]);
+
+        EVP_CIPHER_CTX_free(ctx);
+    }
+
+    printf(resultFmt, passed);
+#endif
+}
+static void test_wolfSSL_EVP_CIPHER_CTX_key_length(void)
+{
+#if defined(OPENSSL_ALL) && !defined(NO_DES3)
+    byte key[AES_256_KEY_SIZE] = {0};
+    byte iv[AES_BLOCK_SIZE] = {0};
+
+    printf(testingFmt, "wolfSSL_EVP_CIPHER_CTX_key_length");
+
+    EVP_CIPHER_CTX *ctx = EVP_CIPHER_CTX_new();
+    const EVP_CIPHER *init = EVP_des_ede3_cbc();
+
+    wolfSSL_EVP_CIPHER_CTX_init(ctx);
+    AssertIntEQ(EVP_CipherInit(ctx, init, key, iv, 1), WOLFSSL_SUCCESS);
+
+    AssertIntEQ(wolfSSL_EVP_CIPHER_CTX_key_length(ctx), 24);
+
+    EVP_CIPHER_CTX_free(ctx);
+    printf(resultFmt, passed);
+#endif
+}
+static void test_wolfSSL_EVP_CIPHER_CTX_set_key_length(void)
+{
+#if defined(OPENSSL_ALL) && !defined(NO_DES3)
+    byte key[AES_256_KEY_SIZE] = {0};
+    byte iv[AES_BLOCK_SIZE] = {0};
+    int keylen;
+
+    printf(testingFmt, "wolfSSL_EVP_CIPHER_CTX_set_key_length");
+
+    EVP_CIPHER_CTX *ctx = EVP_CIPHER_CTX_new();
```

Comment:
We try to declare all variables at the top of a function.  Can you move the declaration of "EVP_CIPHER_CTX* ctx = NULL" to the top of the function, then only assign/initialize it here?  Same applies to "const EVP_CIPHER* init" on the line below.